users/vieting/jobs/nbest.py

```diff
@@ -0,0 +1,117 @@
+    ):
+        kwargs = locals()
+        del kwargs["self"]
+
+        self.config, self.post_config = self.create_config(**kwargs)
+        self.concurrent = crp.concurrent if parallelize else 1
+        self.exe = self.select_exe(crp.flf_tool_exe, "flf-tool")
+        self.lattice_cache = lattice_cache
+
+        self.out_log_file = self.log_file_output_path("nbest_from_lattice", crp, self.concurrent > 1)
+        self.out_nbest_file = self.output_path("nbest")
+
+        self.rqmt = {
+            "time": max(crp.corpus_duration / (5.0 * self.concurrent), 0.5),
+            "cpu": 1,
+            "mem": 4,
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt, args=range(1, self.concurrent + 1))
+        if self.concurrent > 1:
+            yield Task("merge", mini_task=True, tries=2)  # 2 tries to catch fs problems crashing the job
+
+    def create_files(self):
+        self.write_config(self.config, self.post_config, "nbest_from_lattice.config")
+        self.write_run_script(self.exe, "nbest_from_lattice.config")
+
+    def run(self, task_id):
+        log_file = self.out_log_file if self.concurrent <= 1 else self.out_log_file[task_id]
+        self.run_script(task_id, log_file)
+        if self.concurrent <= 1:
+            shutil.move("nbest.1", self.out_nbest_file.get_path())
+
+    def merge(self):
+        with open(self.out_nbest_file.get_path(), "wt") as out:
+            for t in range(1, self.concurrent + 1):
+                assert os.path.getsize("nbest.%d" % t) > 0, "Empty File nbest.%d, maybe restart merge" % t
+                with open("nbest.%d" % t, "rt") as f:
+                    shutil.copyfileobj(f, out)
+
+    @classmethod
+    def create_config(
+        cls,
+        crp,
+        lattice_cache,
+        n,
+        parallelize,
+        extra_config,
+        extra_post_config,
+        **kwargs,
+    ):
+        config, post_config = rasr.build_config_from_mapping(
+            crp,
+            {
+                "corpus": "flf-lattice-tool.corpus",
+                "lexicon": "flf-lattice-tool.lexicon",
+            },
+            parallelize=parallelize,
+        )
+        # segment
+        config.flf_lattice_tool.network.initial_nodes = "segment"
+        config.flf_lattice_tool.network.segment.type = "speech-segment"
+        config.flf_lattice_tool.network.segment.links = "0->archive-reader:1 0->dump-n-best:1"
+
+        # read lattice
+        config.flf_lattice_tool.network.archive_reader.type = "archive-reader"
+        config.flf_lattice_tool.network.archive_reader.links = "reduce-scores"
+        config.flf_lattice_tool.network.archive_reader.format = "flf"
+        config.flf_lattice_tool.network.archive_reader.path = lattice_cache
+        config.flf_lattice_tool.network.archive_reader.flf.append.keys = "confidence"
+        config.flf_lattice_tool.network.archive_reader.flf.append.confidence.scale = 0.0
```

Comment:
I see you appending a key "confidence" here, but it does not seem to be used later..?
```suggestion
```